File: An 2/Semestrul 1/Metode Avansate de Programare/Lab 2/Sem1-MAP/Sem1-MAP/src/StrategyTaskRunner.java
public class StrategyTaskRunner implements TaskRunner{
    private Container container;

    public StrategyTaskRunner(Strategy strategy) {
        TaskContainerFactory factory=TaskContainerFactory.getInstance();
        container= factory.createContainer(strategy);
    }

    @Override
    public void executeOneTask() {
        Task t=container.remove();
        t.execute();
    }

    @Override
    public void executeAll() {
        while(hasTask()){
            executeOneTask();
        }
    }

    @Override
    public void addTask(Task t) {
        container.add(t);
    }

    @Override
    public boolean hasTask() {
        return !container.isEmpty();
    }
}

File: An 2/Semestrul 1/Metode Avansate de Programare/Lab 9/lab9_map/lab9_map/StrategyTaskRunner.cs
namespace lab9_map;

public class StrategyTaskRunner : TaskRunner
{
    private readonly Container container;
    
    public StrategyTaskRunner(Strategy strategy)
    {
        TaskContainerFactory factory = TaskContainerFactory.Instance;
        container = factory.CreateContainer(strategy);
    }
    
    public void ExecuteOneTask()
    {
        Task t = container.Remove();
        t.Execute();
    }
    
    public void ExecuteAll()
    {
        while (HasTask())
        {
            ExecuteOneTask();
        }
    }
    
    public void AddTask(Task t)
    {
        container.Add(t);
    }
    
    public bool HasTask()
    {
        return !container.IsEmpty();
    }
}